src/app/components/bibliography-info/biblio-styled/biblio-styled.component.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { AfterViewInit, ChangeDetectorRef, Component, Input, OnChanges, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
+import { AppConfig, BibliographicStyle, CitingRanges } from 'src/app/app.config';
+import { BibliographicEntry, BibliographicStructEntry } from 'src/app/models/evt-models';
+
+@Component({
+  selector: 'evt-styled-biblio-entry',
+  templateUrl: './biblio-styled.component.html',
+  styleUrls: ['./biblio-styled.component.scss'],
+})
+export class StyledBiblioEntryComponent implements OnChanges, AfterViewInit {
+
+  @ViewChild('title', { static: false }) title: TemplateRef<any>;
+  @ViewChild('author', { static: false }) author: TemplateRef<any>;
+  @ViewChild('publication', { static: false }) publication: TemplateRef<any>;
+  @ViewChild('editor', { static: false }) editor: TemplateRef<any>;
+  @ViewChild('date', { static: false }) date: TemplateRef<any>;
+  @ViewChild('pubPlace', { static: false }) pubPlace: TemplateRef<any>;
+  @ViewChild('publisher', { static: false }) publisher: TemplateRef<any>;
+  @ViewChild('doi', { static: false }) doi: TemplateRef<any>;
+
+
+  @Input() data: BibliographicEntry | BibliographicStructEntry;
+  @Input() style: string = AppConfig.evtSettings.ui.defaultBibliographicStyle;
+
+  public biblEntry;
```

Comment:
This will be tricky to change, as I'm using a dynamic assignment to make biblEntry either be a BibliographicEntry or BibliographicStructEntry, but if it's needed to understand the code I'll refactor it.

Reply:
as a suggestion you can declare as  public biblEntry: BibliographicEntry | BibliographicStructEntry;
if it doesn't work decalare as any